server/services/token/token_service.go

```diff
@@ -4,40 +4,29 @@ import (
 	"fmt"
 	"main/server/db"
 	"main/server/model"
-	"main/server/response"
-	"main/server/utils"
 	"os"
 	"time"
 
-	"github.com/gin-gonic/gin"
 	"github.com/golang-jwt/jwt/v4"
 )
 
-type Claims struct {
-	UserId string `json:"userId"`
-	Phone  string `json:"phone"`
-	Role   string `json:"role"`
-	jwt.RegisteredClaims
-}
-
 // Generate JWT Token
-func GenerateToken(claims Claims, ctx *gin.Context) *string {
+func GenerateToken(claims model.Claims) (*string, error) {
 	//create user claims
 
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
 
 	tokenString, err := token.SignedString([]byte(os.Getenv("JWTKEY")))
 
 	if err != nil {
-		response.ShowResponse("Error signing token", utils.HTTP_UNAUTHORIZED, "Failure", nil, ctx)
-		return nil
+		return nil, err
```

Comment:
Check the error message and make sure we are not sending any info related to `JWTKEY` config with it